### PR TITLE
adding how to guide for new adapters

### PR DESCRIPTION
**Pull request by MarkWolters**
New how-to guide for adapter implementation.

### CODE REVIEW

**Review by MikeYaacoubStax — CHANGES_REQUESTED** [review]
I proof-red the guide, it is very helpful and very good work. I kind of wish I had it when creating my first adapter.

One change request, when rendering the site, the indentations of sub-paragraphs was lost, and the numbering got confusing.

Quick image as demonstration:
image

File: site/content/dev-guide/how-tos/implement-an-adapter.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
++++
+title="Implement an Adapter"
+weight=56
++++
+
+NoSqlBench is composed of a sound architectural underpinning that makes the development and addition of new adapters as seamless and quick a process as possible. nb provides a hierarchy of classes which can be extended and interfaces to be implemented, in order to avoid having to re-implement the application infrastructure and in so doing reduces both the complexity and demands on time involved in implementing a new adapter type. However, it is necessary to understand this existing architecture in order to effectively use the provided building blocks. The following is intended as a guide to developing new adapters to extend the functionality of nb5.
```
Comment (jeffbanks) [inline site/content/dev-guide/how-tos/implement-an-adapter.md]: Possible rewording at point of: "quick a process".
Where 'nb' change to 'nb5'.

File: site/content/dev-guide/how-tos/implement-an-adapter.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+# Prerequisites and Assumptions
+* It is assumed that the developer(s) have familiarized themselves with the  [NoSqlBench core concepts](https://docs.nosqlbench.io/introduction/core-concepts/) provided in the nosqlbench documentation.
+* The ability to code in Java is required. NoSqlBench makes extensive use of lambda functionality. It is a requirement to understand and be comfortable using these principles to develop an adapter.
+* This guide is not specific to any particular client. It is assumed that the developer(s) have familiarity with the client they are developing an adapter for.
+
+# Internal Dependencies
+The nb project as a whole is composed of numerous modules and packages. For the purpose of customer adapter implementation there are only two required; adapters-api and nb-annotations.
+
+	<dependency>
```
Comment (jeffbanks) [inline site/content/dev-guide/how-tos/implement-an-adapter.md:16]: Formatting of xml will help the reader.

File: site/content/dev-guide/how-tos/implement-an-adapter.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        <groupId>io.nosqlbench</groupId>
+        <artifactId>nb-annotations</artifactId>
+        <version>${revision}</version>
+        <scope>compile</scope>
+	</dependency>
+	
+The adapters-api and nb-annotations  modules contain all of the base classes, interfaces and annotations required to implement a new adapter and flesh out its functionality.
+
+* Adapters - Every new adapter will, of course, include an Adapter class. This class will be a descendent of the BaseDriverAdapter class found here. Nosqlbench uses the Java Service Provider Interface(SPI) to define service providers. If you don’t know what that is, don't worry about it now. The tl;dr is that extending the BaseDriverAdapter and annotating your driver adapter class with the Service annotation will result in that class being added to META-INF/services under the specified class name, making it available as a service provider. Nosqlbench takes care of all the rest!
```
Comment (jeffbanks) [inline site/content/dev-guide/how-tos/implement-an-adapter.md]: Referenced the BaseDriverAdapter class as found here.  There needs to be a link included.

File: site/content/dev-guide/how-tos/implement-an-adapter.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+* Ops - Nosqlbench is built around the ability to define operations (“Ops”) in a generic fashion (the op template) and translate them into actions. All of the basic plumbing for this is handled in the adapters-api module, from the yaml file loader to the definition of the various types of ops available to implement. Along with the Op types the adapters-api package defines the base classes and interfaces for Op Dispensers, which do the work of creating the Op, and the Op Mappers, which map the appropriate functionality based on the Op type presented. We’re going to get into the details of the available implementations in the next section.
```
Comment (jeffbanks) [inline site/content/dev-guide/how-tos/implement-an-adapter.md]: NoSQLBench

File: site/content/dev-guide/how-tos/implement-an-adapter.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+# Your First Adapter
```
Comment (jeffbanks) [inline site/content/dev-guide/how-tos/implement-an-adapter.md:35]: Formatting and update of numbering needed in the section below.

File: site/content/dev-guide/how-tos/implement-an-adapter.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+1. Familiarize yourself with the native driver. You will need to know exactly what functionality you want to test and what the APIs for that functionality look like so you can plan for what your Op classes need to be able to do.
+1. Create a new module in the root of the project using standard naming:
+	2. 	adapter-*\<typeName>* where the chosen selector that will be used with driver=*\<typeName>*
+		1. e.g. adapter-jdbc, adapter-pinecone, etc.
+	2. package naming should follow io.nosqlbench.adapter.*\<typeName>*
+	1. A quick way to create module is to copy an existing adapter's pom.xml into a new directory as pod.xml, modify the naming elements, and then rename pod.xml to pom.xml
+		1. In most cases the only dependencies you should need are adapters-api and nb-annotations (as discussed in the previous section), as well as the api library published for the native driver you are adapting. If you find that you need to pull in additional dependencies, first verify whether they are included in another module already within the nb project to simplify the process.
```
Comment (jeffbanks) [inline site/content/dev-guide/how-tos/implement-an-adapter.md]: +1 I like the comment to "verify other modules"

**Review by jeffbanks — CHANGES_REQUESTED** [review]
Great content here with detail.  

A few items for consideration and fixing.  

A future consideration - an applied usage of this content w/ the Pinecone adapter in a blog on our site.

File: site/content/dev-guide/how-tos/implement-an-adapter.md (hunk 1 above)
Comment (MarkWolters) [reply site/content/dev-guide/how-tos/implement-an-adapter.md]: Wording changed to "...makes the process of implementing new adapters as seamless and quick as possible". nb5 --> nb.

File: site/content/dev-guide/how-tos/implement-an-adapter.md (hunk 2 above)
Comment (MarkWolters) [reply site/content/dev-guide/how-tos/implement-an-adapter.md:16]: sorry, not clear what is meant here. This section appears formatted correctly to me, am I missing something?

**Review by MikeYaacoubStax — APPROVED** [review]
Indentation issues fixed, LGTM!

File: site/content/dev-guide/how-tos/implement-an-adapter.md (hunk 3 above)
Comment (MarkWolters) [reply site/content/dev-guide/how-tos/implement-an-adapter.md]: updated doc with links wherever an existing class is mentioned.

File: site/content/dev-guide/how-tos/implement-an-adapter.md (hunk 4 above)
Comment (MarkWolters) [reply site/content/dev-guide/how-tos/implement-an-adapter.md]: Corrected all occurrences

File: site/content/dev-guide/how-tos/implement-an-adapter.md (hunk 5 above)
Comment (MarkWolters) [reply site/content/dev-guide/how-tos/implement-an-adapter.md:35]: corrected

File: site/content/dev-guide/how-tos/implement-an-adapter.md (hunk 6 above)
Comment (MarkWolters) [reply site/content/dev-guide/how-tos/implement-an-adapter.md]: Thanks!

**Review by jshook — APPROVED** [review]
Pulling these in so we can preview on builddocs.nosqlbench.io

**Pull request merged**